vortex-layout/src/scan/mod.rs

```diff
@@ -138,6 +153,20 @@ impl ScanBuilder {
 
         let row_masks = splits
             .into_iter()
+            .filter_map(|row_range| {
+                if let Some(scan_range) = &self.row_range {
+                    // If the row range is fully within the scan range, return it.
+                    if row_range.start < scan_range.start || row_range.end > scan_range.end {
```

Comment:
the logic here is inverted, think it should be:

<img width="801" alt="image" src="https://github.com/user-attachments/assets/b4df1fcf-bae1-44dc-ad3f-872f88b95244" />


that gets us correct results on TPC-H 100